File: android-xml-sample/app/src/main/java/com/nativedisplay/sample/xml/ui/FeedItem.kt
package com.nativedisplay.sample.xml.ui

import com.clevertap.android.nativedisplay.models.ResolvedConfig
import com.nativedisplay.sample.xml.data.Product

sealed class FeedItem {
    abstract val id: String
}

data class NativeProduct(
    override val id: String,
    val product: Product
) : FeedItem()

data class SDUIProduct(
    override val id: String,
    val config: ResolvedConfig
) : FeedItem()

data class SDUIGallery(
    override val id: String,
    val config: ResolvedConfig
) : FeedItem()

File: NativeDisplayUiKit/NativeDisplayUiKit/FeedItem.swift
//
//  FeedItem.swift
//  NativeDisplayUiKit
//
//  Feed item types for mixed native/SDUI content
//

import Foundation
import CleverTapNativeDisplay

enum FeedItem {
    case nativeProduct(id: String, product: Product)
    case sduiProduct(id: String, config: ResolvedConfig)
    case sduiGallery(id: String, config: ResolvedConfig)
    
    var id: String {
        switch self {
        case .nativeProduct(let id, _): return id
        case .sduiProduct(let id, _): return id
        case .sduiGallery(let id, _): return id
        }
    }
}
